### PR TITLE
refactor(components/inline-form): inline-form strict mode support

### PR DESCRIPTION
**Pull request by bradenbiz**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/623/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/623/storybooks/ag-grid-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/623/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/623/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/623/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/623/playground/)

File: libs/components/inline-form/src/lib/modules/inline-form/fixtures/inline-form.fixture.ts [hunk 1]
```diff
@@ -20,5 +20,7 @@ export class SkyInlineFormFixtureComponent {
 
   public showFormWithNoElements = false;
 
-  public onClose(event: SkyInlineFormCloseArgs): void {}
+  public onClose(event: SkyInlineFormCloseArgs): void {
+    console.log('sky-inline-form onClose event = ', event);
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/inline-form/src/lib/modules/inline-form/fixtures/inline-form.fixture.ts]: Lets just `return;` and disable the event warning.

Comment (bradenbiz) [reply libs/components/inline-form/src/lib/modules/inline-form/fixtures/inline-form.fixture.ts]: done in: [remove console.log. ignore eslint warning](https://github.com/blackbaud/skyux/pull/623/commits/9c5a45c7e29e6a89b3ed96176d3a50bd4bb56496)

File: libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts [hunk 2]
```diff
@@ -88,22 +88,34 @@ export class SkyInlineFormComponent implements OnInit, OnDestroy {
   // eslint-disable-next-line @angular-eslint/no-output-native
   public close = new EventEmitter<SkyInlineFormCloseArgs>();
 
-  public buttons: SkyInlineFormButtonConfig[];
+  public buttons!: SkyInlineFormButtonConfig[];
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts:92]: We should probably account for `button` being undefined.

Comment (Blackbaud-TrevorBurch) [reply libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts:92]: I think we could go that with just one or two null checks and then adding `button` to the `ngIf` on line 6 of the template.

File: libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts [hunk 3]
```diff
@@ -257,7 +273,7 @@ export class SkyInlineFormComponent implements OnInit, OnDestroy {
         break;
     }
 
-    return promise;
+    return promise; // uhhhh not sure how this was working before?? needs to be duplicated across each case statement... right? then needs to have an else statement?
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts]: Lets touch base and chat about this. I think I see how we can make this better.

File: libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts (hunk 2 above)
Comment (bradenbiz) [reply libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts:92]: done in: [remove comment. buttons can be undefined](https://github.com/blackbaud/skyux/pull/623/commits/1e16394ccc73bd3e1cada21693f9ed0e8a987e9c)

File: libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts (hunk 3 above)
Comment (bradenbiz) [reply libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts]: created a separate bug for this. will do in a separate PR

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**